### PR TITLE
Add support for shared VPC

### LINKED ISSUE #452: Ability to associate a service project with a Shared VPC after the Shared VPC is created
In our setup we want to have separate Terraform state files for a Shared VPC host project and multiple associated service projects.
To do this we need to be able to associate the service project with the Shared VPC after the Shared VPC has been created.
This is similar to the use case for the [`aws_vpc_endpoint_route_table_association`](https://www.terraform.io/docs/providers/aws/r/vpc_endpoint_route_table_association.html) resource.

### PR DESCRIPTION
**Pull request by rosbo**
Fixes #452 

Introduced two new sources:
* `google_compute_shared_vpc_host_project`
* `google_compute_shared_vpc_service_project`

```sh
TF_ACC=1 go test ./google -v -run TestAccComputeSharedVpc_basic -timeout 120m
=== RUN   TestAccComputeSharedVpc_basic
--- PASS: TestAccComputeSharedVpc_basic (176.18s)
PASS
```

### CODE REVIEW

File: google/resource_compute_shared_vpc_service_project.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	serviceProject := d.Get("service_project").(string)
+
+	req := config.clientCompute.Projects.GetXpnResources(hostProject)
+	if err := req.Pages(context.Background(), func(page *compute.ProjectsGetXpnResources) error {
+		for _, xpnResourceId := range page.Resources {
+			if xpnResourceId.Type == "PROJECT" && xpnResourceId.Id == serviceProject {
+				return nil
+			}
+		}
+		return fmt.Errorf("%s is not a service project of %s", serviceProject, hostProject)
```
Comment (danawillow) [inline google/resource_compute_shared_vpc_service_project.go]: What if the result is on the second page of results?

File: google/resource_compute_shared_vpc_service_project.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	}
+	if err = computeOperationWait(config, op, hostProject, "Disabling Shared VPC Resource"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func isDisabledXpnResourceError(err error) bool {
+	if gerr, ok := err.(*googleapi.Error); ok {
+		if gerr.Code == 400 && len(gerr.Errors) > 0 && gerr.Errors[0].Reason == "invalidResourceUsage" {
```
Comment (danawillow) [inline google/resource_compute_shared_vpc_service_project.go]: Can you add a comment as to why this error is special?

File: google/resource_compute_shared_vpc_test.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+resource "google_project_services" "service" {
+	project  = "${google_project.service.project_id}"
+	services = ["compute.googleapis.com"]
+}
+
+resource "google_compute_shared_vpc_host_project" "host" {
+	project    = "${google_project.host.project_id}"
+	depends_on = ["google_project_services.host", "google_project_services.service"]
```
Comment (danawillow) [inline google/resource_compute_shared_vpc_test.go]: shouldn't the service project depend on the service apis being enabled, not the host?

File: website/docs/r/compute_shared_vpc_host_project.html.markdown [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
+layout: "google"
+page_title: "Google: google_compute_shared_vpc_host_project"
+sidebar_current: "docs-google-compute-shared-vpc-host-project"
+description: |-
+ Allows enablig and disabling Shared VPC for the host Google Cloud Platform project.
```
Comment (danawillow) [inline website/docs/r/compute_shared_vpc_host_project.html.markdown]: typo: enabling

File: website/docs/r/compute_shared_vpc_host_project.html.markdown [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+---
+
+# google\_compute\_shared\_vpc\_host\_project
+
+Allows enablig and disabling Shared VPC for the host Google Cloud Platform project. For more information see
```
Comment (danawillow) [inline website/docs/r/compute_shared_vpc_host_project.html.markdown]: here too

File: website/docs/r/compute_shared_vpc_service_project.html.markdown [hunk 6]
```diff
@@ -0,0 +1,28 @@
+---
+layout: "google"
+page_title: "Google: google_compute_shared_vpc_service_project"
+sidebar_current: "docs-google-compute-shared-vpc-service-project"
+description: |-
+ Allows enablig and disabling Shared VPC for a service Google Cloud Platform project.
```
Comment (danawillow) [inline website/docs/r/compute_shared_vpc_service_project.html.markdown]: and here and line 11

File: google/resource_compute_shared_vpc_test.go [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"os"
+)
+
+func TestAccComputeSharedVpc_basic(t *testing.T) {
```
Comment (danawillow) [inline google/resource_compute_shared_vpc_test.go]: Can you also check that delete works? https://github.com/terraform-providers/terraform-provider-google/pull/396/commits/5f9ef48338d46fa57680c1e84af19c2e9eadfe08 may be of use here

Comment (rosbo) [reply google/resource_compute_shared_vpc_test.go]: Done

**Comment by rosbo** [comment]
```sh
TF_ACC=1 go test ./google -v -run TestAccComputeSharedVpc_basic -timeout 120m
=== RUN   TestAccComputeSharedVpc_basic
--- PASS: TestAccComputeSharedVpc_basic (164.84s)
PASS
```

**Pull request merged**